adversarial_workflow/utils/citations.py

```
@@ -0,0 +1,608 @@
+def save_cache(cache_path: Path, cache: dict[str, dict]) -> None:
+    """Save URL cache to disk."""
+    with open(cache_path, "w") as f:
+        json.dump(cache, f, indent=2)
+
+
+def get_cache_key(url: str) -> str:
+    """Generate a cache key for a URL."""
+    return hashlib.md5(url.encode()).hexdigest()
+
+
+def classify_response(status_code: int, headers: dict, content: Optional[str] = None) -> URLStatus:
```

Ruff (0.14.14)

[warning] Unused function argument: `headers`

(ARG001)